packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts

```diff
@@ -0,0 +1,176 @@
+import { ModelTransformer } from '@aws-amplify/graphql-model-transformer';
```

Comment:
This ports the V1 transformer tests that covered `@auth` directives on root types. It should cover the same use cases as the V1 test, although I slightly reworded the `it` clauses to more clearly (according to me) reflect what's being asserted.